packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -588,26 +610,247 @@ export class UiControlClient extends ApiCommands {
       }
     },
   ) {
-    let command = this.click().text();
+    let command: FluentFiltersOrRelations | FluentFiltersOrRelationsGetter = this.click().text();
+
+    command = await this.evaluateMatchingProperty(command, params.type, params.text);
+
+    if (params.relation) {
+      command = this.evaluateRelation(command, params.relation.type, params.relation.text);
+    }
+
+    await command.exec();
+  }
 
-    switch (params.type) {
+  // eslint-disable-next-line class-methods-use-this
+  private async evaluateFinalExpectExistenceCommand(
+    finalCommand: FluentFiltersOrRelationsGetter,
+    element: ExpectExistenceInputParameter,
+  ): Promise<ExpectExistenceInputParameter> {
+    const el = element;
+    el.exists = false;
+    if ((await finalCommand.exec()).length > 0) {
+      el.exists = true;
+    }
+    return new Promise<ExpectExistenceInputParameter>((resolve) => {
+      resolve(el);
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private async evaluateMatchingProperty(
+    command: FluentFiltersOrRelationsGetter | FluentFiltersOrRelations,
+    matching: 'similar' | 'exact' | 'regex' | undefined,
+    text: string,
+  ): Promise<FluentFiltersOrRelationsGetter | FluentFiltersOrRelations> {
+    let commando = command;
+    switch (matching) {
       case 'similar':
-        command = command.withText(params.text);
+        commando = command.withText(text);
         break;
       case 'exact':
-        command = command.withExactText(params.text);
+        commando = command.withExactText(text);
         break;
       case 'regex':
-        command = command.withTextRegex(params.text);
+        commando = command.withTextRegex(text);
         break;
       default:
-        throw new ValidationError('"type" must be "similar", "exact" or "regex"');
+        throw new ValidationError('"matching" property must be "similar", "exact" or "regex"');
     }
+    return new Promise<FluentFiltersOrRelationsGetter | FluentFiltersOrRelations>((resolve) => {
+      resolve(commando);
+    });
+  }
 
-    if (params.relation) {
-      command = this.evaluateRelation(command, params.relation.type, params.relation.text);
+  private evaluateRelationExpectExistence(
+    element: ExpectExistenceInputParameter,
+    command: FluentFiltersOrRelationsGetter,
+  ) {
+    if (element.relation === undefined) {
+      throw new ValidationError(
+        '"relation" must never be undefined when searching for a switch or checkbox.',
+      );
     }
+    // param.relation.text must never be undefined here -> User error
```

Comment:
Wrong place for this validation... This method knows more than it should know...